src/buffer/base.ts

```diff
@@ -0,0 +1,425 @@
+   */
+  table(table: string): SenderBuffer {
+    if (typeof table !== "string") {
+      throw new Error(`Table name must be a string, received ${typeof table}`);
+    }
+    if (this.hasTable) {
+      throw new Error("Table name has already been set");
+    }
+    validateTableName(table, this.maxNameLength);
+    this.checkCapacity([table], table.length);
+    this.writeEscaped(table);
+    this.hasTable = true;
+    return this;
+  }
+
+  /**
+   * Writes a symbol name and value into the buffer.
+   *
+   * @param {string} name - Symbol name.
+   * @param {unknown} value - Symbol value, toString() is called to extract the actual symbol value from the parameter.
+   * @return {Sender} Returns with a reference to this sender.
+   */
+  symbol(name: string, value: unknown): SenderBuffer {
+    if (typeof name !== "string") {
+      throw new Error(`Symbol name must be a string, received ${typeof name}`);
+    }
+    if (!this.hasTable || this.hasColumns) {
+      throw new Error(
+        "Symbol can be added only after table name is set and before any column added",
+      );
+    }
+    const valueStr = value.toString();
+    this.checkCapacity([name, valueStr], 2 + name.length + valueStr.length);
+    this.write(",");
+    validateColumnName(name, this.maxNameLength);
+    this.writeEscaped(name);
+    this.write("=");
+    this.writeEscaped(valueStr);
+    this.hasSymbols = true;
+    return this;
+  }
+
+  /**
+   * Writes a string column with its value into the buffer.
+   *
+   * @param {string} name - Column name.
+   * @param {string} value - Column value, accepts only string values.
+   * @return {Sender} Returns with a reference to this sender.
+   */
+  stringColumn(name: string, value: string): SenderBuffer {
+    this.writeColumn(
+      name,
+      value,
+      () => {
+        this.checkCapacity([value], 2 + value.length);
+        this.write('"');
+        this.writeEscaped(value, true);
+        this.write('"');
+      },
+      "string",
+    );
+    return this;
+  }
+
+  /**
+   * Writes a boolean column with its value into the buffer.
+   *
+   * @param {string} name - Column name.
+   * @param {boolean} value - Column value, accepts only boolean values.
+   * @return {Sender} Returns with a reference to this sender.
+   */
+  booleanColumn(name: string, value: boolean): SenderBuffer {
+    this.writeColumn(
+      name,
+      value,
+      () => {
+        this.checkCapacity([], 1);
+        this.write(value ? "t" : "f");
+      },
+      "boolean",
+    );
+    return this;
+  }
+
+  /**
+   * Writes a float column with its value into the buffer.
+   *
+   * @param {string} name - Column name.
+   * @param {number} value - Column value, accepts only number values.
+   * @return {Sender} Returns with a reference to this sender.
+   */
+  abstract floatColumn(name: string, value: number): SenderBuffer;
+
+  /**
+   * Writes an integer column with its value into the buffer.
```

Comment:
nit: we should probably mention that this method writes 64-bit integer and that it works for `long` and smaller integer database types.

Reply:
changed in https://github.com/questdb/nodejs-questdb-client/pull/49/commits/f9b58a4dec6065626e96ffde8fada198969e839c